crates/router/src/lib.rs

```diff
@@ -149,10 +151,41 @@ pub async fn start_server(conf: settings::Settings) -> ApplicationResult<(Server
         .workers(server.workers)
         .shutdown_timeout(server.shutdown_timeout)
         .run();
-
+    tokio::spawn(receiver_for_error(rx, server.handle()));
     Ok((server, app_state))
 }
 
+pub async fn receiver_for_error(rx: oneshot::Receiver<()>, server: impl Stop) {
+    match rx.await {
+        Ok(_) | Err(_) => {
```

Comment:
Just log the error if receiver throws any. We need to stop the server only in the Ok(_) case